apps/authentication/routes.py

```diff
@@ -128,24 +137,74 @@ def register():
 
         # else we can create the user
         user = Users(**request.form)
-        db.session.add(user)
-        db.session.commit()
 
-        return render_template('pages/register.html',
-                               msg='User created please <a href="/login">login</a>',
-                               success=True,
-                               form=create_account_form)
+        confirmation_token = str(uuid.uuid4().int)[:6]
+        session['confirmation_token'] = confirmation_token
+        session['user'] = request.form
+        session['datetime'] = utcnow()
+
+        # Send email
+        msg = Message(
+            subject="HackInSDN confirmation code",
+            sender=app.config['MAIL_USERNAME'],
+            recipients=[email],
+            body=f"This is your confirmation code: {confirmation_token}"
+        )
+        mail.send(msg)
+
+        return redirect(url_for('authentication_blueprint.confirm_page'))
 
     else:
         return render_template('pages/register.html', form=create_account_form)
 
+@blueprint.route('/confirm', methods=['GET', 'POST'])
+def confirm_page():
+    form = ConfirmAccountForm(request.form)
+
+    confirmation_token = session.get('confirmation_token')
+    if not confirmation_token:
+        return redirect(url_for('authentication_blueprint.register'))
+        
+    if 'confirm' in request.form:
+        user = Users(**session.get('user'))
+        created_at = session.get('datetime')
+     
+        now = utcnow()
+        if now - created_at > timedelta(minutes=5):
+            return render_template('pages/confirm.html', msg='Token expired, please <a href=/register>click here</a> to register again', success=False, form=form)
+
+        if request.form['confirmation_token'] != confirmation_token:
+            return render_template('pages/confirm.html', msg='Invalid token', success=False, form=form)
+
+        session.pop('confirmation_token')
+
+        db.session.add(user)
+        db.session.commit()
+
+        return redirect(url_for('authentication_blueprint.login'))
+    
+    return render_template('pages/confirm.html', form=form)
+
+@blueprint.route('/resend-code', methods=['GET', 'POST'])
```

Comment:
It looks like `resend-code` is only called with GET.
```suggestion
@blueprint.route('/resend-code', methods=['GET'])
```